tests/test_dimension.py

```diff
@@ -45,6 +45,38 @@ def test_multi_buffer_long_time(self):
         for i in range(7):
             assert np.allclose(g.data[i], 14+i+1)
 
+    def test_modulo_dims_generation(self):
+        grid = Grid((3, 3))
+
+        f = TimeFunction(name="f", grid=grid, save=Buffer(2))
```

Comment:
Can we add a test that

```
f = TimeFunction(name="f", grid=grid, save=Buffer(2), time_order=2)
f.dt2 == ...
```

It works correctly but would be good to have it as a safety because this `Buffer(2)` will be mostly used in those cases